lib/services/nativescript-project-plugins-service.ts

```diff
@@ -0,0 +1,463 @@
+///<reference path="../.d.ts"/>
+"use strict";
+
+import path = require("path");
+import * as util from "util";
+import {EOL} from "os";
+import Future = require("fibers/future");
+import * as shelljs from "shelljs";
+import * as semver from "semver";
+
+import PluginsDataLib = require("../plugins-data");
+
+export class NativeScriptProjectPluginsService implements IPluginsService {
+	private static MESSAGES = ["NPM Packages", "NPM NativeScript Plugins", "Marketplace Plugins"];
+
+	constructor(
+		private $errors: IErrors,
+		private $logger: ILogger,
+		private $project: Project.IProject,
+		private $projectConstants: Project.IProjectConstants,
+		private $fs: IFileSystem,
+		private $server: Server.IServer,
+		private $httpClient: Server.IHttpClient,
+		private $childProcess: IChildProcess) {
+		}
+
+	public getAvailablePlugins(pluginsCount?: number): IPlugin[] {
+		let count = pluginsCount || 10;
+		let futures = [
+			this.getUniqueMarketplacePlugins(),
+			this.getTopNpmPackages(count),
+			this.getTopNativeScriptNpmPackages(count)
+		];
+		Future.wait(futures);
+		let results = _(futures)
+						.map(f => f.get())
+						.filter(res => !!res)
+						.flatten<IPlugin>()
+						.value();
+		return results;
+	}
+
+	public findPlugins(keywords: string[]): IFuture<IBasicPluginInformation[]> {
+		return ((): IBasicPluginInformation[] => {
+			let findPluginByNameFutures = _.map(keywords, keyword => this.findPluginsByName(keyword));
+			Future.wait(findPluginByNameFutures);
+			let results = _(findPluginByNameFutures)
+							.map(f => f.get())
+							.filter(plugins => !!plugins && !!plugins.length)
+							.flatten<IBasicPluginInformation>()
+							.value();
+			return results;
+		}).future<IBasicPluginInformation[]>()();
+	}
+
+	public getInstalledPlugins(): IPlugin[] {
+		let pathToPackageJson = path.join(this.$project.getProjectDir().wait(), this.$projectConstants.PACKAGE_JSON_NAME);
+
+		if(this.$fs.exists(pathToPackageJson).wait()) {
+			let content = this.$fs.readJson(pathToPackageJson).wait();
+			if(content && content.dependencies) {
+				let items =  _.map(content.dependencies, (version: string, name: string) => {
+					let marketplacePlugin = _.find(this.getMarketplacePlugins().wait(), pl => pl.data.Name === name && pl.data.Version === version);
+					return marketplacePlugin || this.getDataForNpmPackage(name, version).wait();
+				}).filter(i => !!i);
+
+				return items;
+			}
+		}
+
+		return null;
+	}
+
+	public printPlugins(plugins: IPlugin[]): void {
+		let groups = _.groupBy(plugins, (plugin: IPlugin) => plugin.type);
```

Comment:
this method shares the same code with CordovaProjectPluginsService, consider moving it to PluginsService